registration-processor/core-processor/registration-processor-abis-handler-stage/src/test/java/io/mosip/registration/processor/abis/handler/stage/test/AbisHandlerStageTest.java

```diff
@@ -1,500 +1,738 @@
+
+		Mockito.when(env.getProperty("DATASHARECREATEURL")).thenReturn("/v1/datashare/create");
+		AbisApplicationDto dto = new AbisApplicationDto();
+		dto.setCode("ABIS1");
+		abisApplicationDtos.add(dto);
+
+		Mockito.doNothing().when(description).setMessage(any());
+		Mockito.when(description.getMessage()).thenReturn("description");
+
+		byte[] bdbBytes = new byte[1048];
+
+		List<BIR> birTypeList = new ArrayList<>();
+		BIR birType1 = new BIR.BIRBuilder().build();
+		BDBInfo bdbInfoType1 = new BDBInfo.BDBInfoBuilder().build();
+		io.mosip.kernel.biometrics.entities.RegistryIDType registryIDType = new RegistryIDType();
+		registryIDType.setOrganization("Mosip");
+		registryIDType.setType("257");
+		io.mosip.kernel.biometrics.constant.QualityType quality = new QualityType();
+		quality.setAlgorithm(registryIDType);
+		quality.setScore(90l);
+		bdbInfoType1.setQuality(quality);
+		BiometricType singleType1 = BiometricType.FINGER;
+		List<BiometricType> singleTypeList1 = new ArrayList<>();
+		singleTypeList1.add(singleType1);
+		List<String> subtype1 = new ArrayList<>(Arrays.asList("Left", "RingFinger"));
+		bdbInfoType1.setSubtype(subtype1);
+		bdbInfoType1.setType(singleTypeList1);
+		birType1.setBdbInfo(bdbInfoType1);
+		birType1.setBdb(bdbBytes);
+		birTypeList.add(birType1);
+
+		BiometricRecord biometricRecord = new BiometricRecord();
+		biometricRecord.setSegments(birTypeList);
+		when(utility.getDefaultSource(any(), any())).thenReturn("reg-client");
+		when(cbeffutil.createXML(any())).thenReturn("abishandlerstage".getBytes());
+
+		Mockito.when(packetManagerService.getBiometrics(any(), any(), any(), any(), any())).thenReturn(biometricRecord);
+
+		Mockito.doNothing().when(registrationStatusDto).setLatestTransactionStatusCode(any());
+		Mockito.doNothing().when(registrationStatusService).updateRegistrationStatus(any(), any(), any());
+
+		Mockito.when(packetInfoManager.getAbisRequestsByBioRefId(any())).thenReturn(abisRequestDtoList);
+
+		AbisQueueDetails abisQueueDetails = new AbisQueueDetails();
+		abisQueueDetails.setName("ABIS1");
+		List<AbisQueueDetails> abisQueueDetailsList = new ArrayList<>();
+		abisQueueDetailsList.add(abisQueueDetails);
+		Mockito.when(utility.getAbisQueueDetails()).thenReturn(abisQueueDetailsList);
+
+		AuditResponseDto auditResponseDto = new AuditResponseDto();
+		ResponseWrapper<AuditResponseDto> responseWrapper = new ResponseWrapper<>();
+		responseWrapper.setResponse(auditResponseDto);
+		Mockito.when(auditLogRequestBuilder.createAuditRequestBuilder(any(), any(), any(), any(), any(), any(), any()))
+				.thenReturn(responseWrapper);
+
+		DataShareResponseDto dataShareResponseDto = new DataShareResponseDto();
+		DataShare dataShare = new DataShare();
+		dataShare.setUrl("http://localhost");
+		dataShareResponseDto.setDataShare(dataShare);
+
+		mockDataSharePolicy(Lists.newArrayList(BiometricType.FACE, BiometricType.FINGER));
+		setMetaInfoMap(new LinkedList<String>(getExceptionModalityMap().values()));

```

Comment:
Why all the segments are marked as exception as per the setup method ?

Reply:
Incorporated review comments